tuxemon/event/actions/npc_look.py

```diff
@@ -0,0 +1,96 @@
+# SPDX-License-Identifier: GPL-3.0
+# Copyright (c) 2014-2023 William Edwards <shadowapex@gmail.com>, Benjamin Bean <superman2k5@gmail.com>
+from __future__ import annotations
+
+import itertools
+import random
+from dataclasses import dataclass
+from typing import Optional, final
+
+from tuxemon.event import get_npc
+from tuxemon.event.eventaction import EventAction
+from tuxemon.map import Direction
+from tuxemon.npc import NPC
+from tuxemon.states.world.worldstate import WorldState
+
+
+@final
+@dataclass
+class NpcLookAction(EventAction):
+    """
+    Make an NPC look around.
+
+    Script usage:
+        .. code-block::
+
+            npc_look <npc_slug>[,frequency][,directions]
+
+    Script parameters:
+        npc_slug: Either "player" or npc slug name (e.g. "npc_maple").
+        frequency: Frequency of movements. 0 to stop looking. If set to
+            a different value it will be clipped to the range [0.5, 5].
+            If not passed the default value is 1.
+        directions: the direction the npc is going to look, by default
+            all
+
+        eg. npc_look npc_slug
+        eg. npc_look npc_slug,,right:left
+
+    """
+
+    name = "npc_look"
+    npc_slug: str
+    frequency: Optional[float] = None
+    directions: Optional[str] = None
+
+    def start(self) -> None:
+        player = self.session.player
+        npc = get_npc(self.session, self.npc_slug)
+        world = self.session.client.get_state_by_name(WorldState)
+        self.limit_direction = []
+        if self.directions:
+            self.limit_direction = self.directions.split(":")
+
+        def _look(npc: NPC) -> None:
+            # Suspend looking around if a dialog window is open
+            for state in self.session.client.active_states:
+                if state.name == "DialogState":
+                    # retrieve NPC talking to
+                    if player.facing == "down":
+                        return
+                    elif player.facing == "up":
+                        return
+                    elif player.facing == "left":
+                        return
+                    elif player.facing == "right":
+                        return
```

Comment:
I don't get it, why put all these conditions in different blocks if at the end the consequence is always the same?

Change suggestion:
```suggestion
                    if player.facing in ("down", "up", "left", "right"):
                        return
```

Btw, aren't `down`, `up`, `left `and `right `all the possible facing the player can take? If so, it means the condition is always `true`... At this point, better just always `return`.

Reply:
you're right

Reply:
fixed a couple of more stuff, removing some unused import, setting list(str) and casting at the end as Direction because npc.facing accepts Direction (mypy typehints)